nano/node/blockprocessor.hpp

```diff
@@ -38,6 +38,26 @@ enum class block_source
 std::string_view to_string (block_source);
 nano::stat::detail to_stat_detail (block_source);
 
+class block_processor_config final
```

Comment:
This could be its own file. Including node_config.hpp pulls in blockprocessor.hpp -> fair_queue.hpp -> transport/channel.hpp

Reply:
I used this pattern of declaring config next to the class in quite a few places at this point. I find it very convenient when working, to the point where having to always split it and keep in separate file would probably outweigh any compilation speed benefits, at least on my hardware.

But this can be fixed at the nodeconfig level much easier, by just using the same pattern as in node header itself: forward declare and store unique_ptrs to configs with an additional reference for convenience.